src/main/java/edu/hm/hafner/analysis/parser/PolyspaceParser.java

```diff
@@ -0,0 +1,104 @@
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+import static edu.hm.hafner.util.StringContainsUtils.*;
+
+/**
+ * A parser for Polyspace Bug Finder and Code Prover results.
+ * Used for .csv files generated from Bugfinder and CodeProver tools
+ *
+ * @author Eva Habeeb
+ */
+public class PolyspaceParser extends IssueParser {
+
+    private static final long serialVersionUID = -1251248150596418714L;
+
+    /**
+     * Creates a new instance of {@link PolyspaceParser}.
+     */
+    public PolyspaceParser() { }
+
+    @Override
+    public Report parse(final ReaderFactory reader) throws ParsingException {
+        try (Stream<String> lines = reader.readStream()) {
+            return parse(lines);
+        }
+        catch (UncheckedIOException e) {
+            throw new ParsingException(e);
+        }
+    }
+
+    private Report parse(final Stream<String> lines) {
+
+        try (IssueBuilder builder = new IssueBuilder()) {
+            int lineNumber;
+            int colNumber;
+            int limit;
+            Report report = new Report();
+            Iterator<String> lineIterator = lines.iterator();
+
+            while (lineIterator.hasNext()) {
+                String line = lineIterator.next();
+                //System.out.println(line);
+                /* Checks whether "CWE" field is found, which defines the difference between
+                 a BugFinder file and a CodeProver report */
+                if (line.contains("CWE")) {
+                    // BugFinder result file has 16 columns
+                    limit = 16;
+                    lineNumber = 14;
+                    colNumber = 15;
+                }
+                else {
+                    // CodePRover file has 15 columns
+                    limit = 15;
+                    lineNumber = 13;
+                    colNumber = 14;
+                }
+
+                String[] arrOfStr = line.split("\\t", limit);
+                builder.setFileName(arrOfStr[8]);
+                builder.setCategory(arrOfStr[2]);
+                builder.setDescription(arrOfStr[1]);
+                builder.setMessage("Check: " + arrOfStr[5] + " " + arrOfStr[6]);
+                builder.setModuleName(arrOfStr[7]);
+                builder.setColumnStart(arrOfStr[colNumber]);
+                builder.setLineStart(arrOfStr[lineNumber]);
+                builder.setSeverity(mapPriority(arrOfStr));
+
+                report.add(builder.build());
+            }
+            return report;
+        }
+    }
+
+    private Severity mapPriority(final String[] arrOfStr) {
+
+        Severity priority = Severity.WARNING_NORMAL;
+
+        if (equalsIgnoreCase(arrOfStr[10], "Unset") && containsAnyIgnoreCase(arrOfStr[1], "Defect")
+                || equalsIgnoreCase(arrOfStr[10], "Unset") && containsAnyIgnoreCase(arrOfStr[3], "Red")
```

Comment:
As addition to my previous comment:
is `equalsIgnoreCase(arrOfStr[10], "Unset") && containsAnyIgnoreCase(arrOfStr[3], "Red")` for one tool and `equalsIgnoreCase(arrOfStr[10], "High")`  for the other tool? Then it really makes sense to use two parsers.

Reply:
No, this is done for both tools since they both generate the same information. These conditions are to set some kind of priority to the condition that will set the level of severity.